lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -1336,7 +1340,69 @@ class MongoClientInterface {
     }
 
     /**
-     * internal listing function
+     * internal listing function for buckets in v0
+     * @param {String} bucketName bucket name
+     * @param {Object} params internal listing params returned by listing extension
+     * @param {Object} extension listing extention
+     * @param {Object} log logger
+     * @param {Function} cb callback
+     * @return {undefined}
+     */
+    internalListObjectV0(bucketName, params, extension, log, cb) {
+        const c = this.getCollection(bucketName);
+        const stream = new MongoReadStream(c, params, params.mongifiedSearch);
+        const skip = new Skip({
+            extension,
+            gte: params.gte,
+        });
+        let cbDone = false;
```

Comment:
```suggestion
        const cbOnce = jsutil.once(callback);
```

Do you think we can use the builtin helper `once` here? It will manage for you the callback.
See https://github.com/scality/Arsenal/blob/development/8.1/lib/jsutil.js

Reply:
yeah it'll make the code a bit cleaner, good idea